File: src/GoonEngine/ecs/components/tagComponent.c
#include <GoonEngine/gnpch.h>
#include <GoonEngine/ecs/components/tagComponent.h>

TagComponent *gnTagComponentNew()
{
    TagComponent *tagComponent = calloc(1, sizeof(*tagComponent));
    return tagComponent;
}
int gnTagComponentAddTag(TagComponent *component, const char *tag)
{
    component->tags = realloc(component->tags, sizeof(char *) * component->tagCount + 1);
    component->tags[component->tagCount++] = tag;
    return true;
}

int gnTagComponentHasTag(TagComponent *component, const char *tag)
{
    printf("Searching for tag %s\n", tag);
    for (int i = 0; i < component->tagCount; i++)
    {
        if (component->tags[i] == tag)
        {
            printf("Found it\n");

            return true;
        }
    }
    printf("No findy\n");
    return false;
}
const char **gnTagComponentGetTags(TagComponent *component)
{
    return component->tags;
}

File: GoonWorld/GoonEngine/components/TagComponent.cs
namespace GoonEngine;
public class TagComponent : Component
{
    // public TagComponent(IntPtr context, params string[] tags)
    public TagComponent(params string[] tags)
    {
        _componentDataPointer = Api.Components.TagComponent.gnTagComponentNew();
        _ecsComponentDataPtr = ECS.NewComponent(ComponentTypes.TAG_COMPONENT, ComponentPtr);
        foreach (var tag in tags)
        {
            AddTag(tag);
        }
    }
    public bool HasTag(string tag) => Api.Components.TagComponent.gnTagComponentHasTag(ComponentPtr, tag);
    public bool AddTag(string tag) => Api.Components.TagComponent.gnTagComponentAddTag(ComponentPtr, tag);
}